### ISSUE TITLE
feat(progress-bar): Missing `--mdc-linear-progress-active-indicator-shape` property for progress-bar

### ISSUE THREAD

**Issue by uus-dev** [angular/components#28554]
### Feature Description

Currently if I want to change the shape of the progress bar to be round like, I only have the `--mdc-linear-progress-track-shape` to set the track shape, but not option to the the shape of the active-indicator, which will still be square. 
Also, only setting the `--mdc-linear-progress-track-shape`, will result in the the active indicator to overflow, leaving me with an inconsistent progress-bar (see 1st picture)

Hence that leaves me with the better option of just setting the `border-radius` of the `mat-progress-bar` element itself, because this doesn't cause the active-indicator to overflow. In this case the `--mdc-linear-progress-track-shape` becomes redundant.

1. Screenshot with `--mdc-linear-progress-track-shape: 4px;`, where active-indicator overflows:

<img width="201" alt="image" src="https://github.com/angular/components/assets/15202984/13ada9e9-d6e6-48e6-bca8-143cfe47a469">

2. Screenshot with `--mdc-linear-progress-track-shape: 4px;` and `border-radius: 4px`, rendering the `--mdc-linear-progress-track-shape` redundant:

<img width="202" alt="image" src="https://github.com/angular/components/assets/15202984/9e8f4bf0-17ab-4e82-bef0-53945b8573a6">

3. Screenshot with only `border-radius: 4px` on leaving me with an almost round progress bar, except that the active-indicator is not actually round.

<img width="205" alt="image" src="https://github.com/angular/components/assets/15202984/0e29cea1-3d0e-4a6d-8bac-01f4bd431aaf">

So there need to be an option to set the `--mdc-linear-progress-active-indicator-shape` like for other components, so that the active-indicator can be made round, as shown in the [Token and Specs section in the Material 3 specification](https://m3.material.io/components/progress-indicators/specs)

### Use Case

This would enable a very relevant an normal use case that is also shown in the [Token and Specs section in the Material 3 specification](https://m3.material.io/components/progress-indicators/specs).

**Comment by danieljancar** [comment]
Whats the status of this issue? 

```scss
@use '@angular/material' as mat;

  @include mat.progress-bar-overrides((
    active-indicator-height: 10px,
    track-height: 10px,
    track-shape: 100px
  ))
```

![Image](https://github.com/user-attachments/assets/974251f2-11ce-4e0a-b4ca-2daf5d94d5a6)
 
Should defo be fixed pls 🚀❤

**Comment by FruityLeo** [comment]
Yes, it seems like there really is a feature missing because I can't see a use case for changing the appearance of the track when it cannot be matched by the indicator. :(

**Comment by alexandre-lacoche-whoz** [comment]
Whats the status of this issue?

**Comment by RamZal365** [comment]
Is there any progress on this?

**Comment by sarah-alaghbari** [comment]
Agreeing to all comments above: active-track-shape is absolutely essential for creating a rounded corner progress bar without ng-deep or any similar hacky workarounds 🙏
